sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -138,6 +166,14 @@ export interface PipelineRequest {
 
   /** Set to true if the request is sent over HTTP instead of HTTPS */
   allowInsecureConnection?: boolean;
+
+  /**
+   * NODEJS ONLY
+   *
+   * A Node-only option to provide a custom `http.Agent`/`https.Agent`.
+   * Does nothing when running in the browser.
+   */
+  customAgent?: Agent;
```

Comment:
nit: I can decide on which one is better. I like that `customAgent` makes it clear that it overrides the default, but on the other hand it looks that popular NodeJS libraries (request/got/node-fetch) just use `agent` in their options.

Reply:
That's fair. I'm fine changing it to `agent` unless @chradek or @bterlson have any objections